exercicios/para-casa/entregas/ellen-dutra/ContaBancaria.js

```diff
@@ -0,0 +1,41 @@
+class ContaBancaria {
+  constructor(nomeTitular, saldoConta, limiteConta) {
+    this.nomeTitular = nomeTitular;
+    this.saldoConta = saldoConta;
+    this.limiteConta = limiteConta;
+  }
+
+  consultarSaldo() {
+    return this.saldoConta;
+  }
+
+  efetuarSaque(valor) {
+    if (valor <= this.saldoConta) {
+      this.saldoConta -= valor;
+      return this.consultarSaldo();
+    } else if (valor > this.saldoConta && valor < this.limiteConta) {
+      this.limiteConta -= valor;
+      this.saldoConta -= valor;
+      return this.consultarSaldo();
+    } else {
+      throw new Error('Saldo e limite insuficientes');
```

Comment:
Como sugestão, colocaria esta validação no início do if/else, sendo ela a primeira validação do If, porque assim, se cair nesta condição, isso impedirá que as demais validações precisem ser executadas.

Reply:
```suggestion
 if (valor > this.saldoConta + this.limiteConta) {
      throw new Error('Saldo e limite insuficientes');
      }
```